### PR TITLE
docs: Add OAuth 2.0 & JWT how-to

### PR DESCRIPTION
**Pull request by danielgtaylor**
This takes some of the information from #202 and #186 and tries to put together a simple how-to for people wanting to add OAuth 2.0 based auth with JWTs to their API.

### CODE REVIEW

**Review by Insei — COMMENTED** [review]
Great work! It's a nice description!

**Review by sm3142 — COMMENTED** [review]
@danielgtaylor Thanks for the good work. I think this will help a lot of people!

I tried to clear up some of the misconceptions regarding what OAuth2 is (authorization) and isn't (authentication) and OAuth2 being quasi synonymous with JWT (it is not). Fell free to use these suggestions or not - as I mentioned elsewhere, the distinction is not _super_ important for the purpose of the document.

I also added a section regarding different token formats. Since I will have to do just that in the not so distant future (i.e. use opaque access tokens), I may be able to provide an abstract implementation at a later stage.

File: docs/docs/how-to/oauth2-jwt.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+There are three main pieces to using OAuth 2.0 with Huma:
+
+1. Issue an access token to a client application
+2. Document the auth scheme and required permissions
+3. Authorize incoming requests
+
+## Issue an Access Token
+
+Huma does not provide any built-in access token issuing functionality. Instead, you can use any existing library or service to issue tokens. For simplicity's sake, we will assume you are using a third-party service for managing users and issuing tokens, like [Auth0](https://auth0.com/) or [Okta](https://www.okta.com/).A simplified flow chart for OAuth2.0 authorization looks something like this:
```
Comment (sm3142) [inline docs/docs/how-to/oauth2-jwt.md:17]:
```suggestion
Huma does not provide any built-in access token issuing functionality. Instead, you can use any existing library or service to issue tokens. For simplicity's sake, we will assume you are using a third-party service for managing users and issuing tokens, like [Auth0](https://auth0.com/) or [Okta](https://www.okta.com/). A simplified flow chart for OAuth2.0 authorization looks something like this:
```
typo

File: docs/docs/how-to/oauth2-jwt.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+		{"myAuth": {"scope1"}},
+	},
+}, func(ctx context.Context, input *GreetingInput) (*GreetingOutput, error) {
+	// TODO: operation implementation goes here
+	return nil, nil
+})
+```
+
+> [!WARNING] 
+> So far, the code above is only documenting the authorization scheme and required scopes, but does not actually authorize incoming requests. The next section will explain  to achieve the latter.
```
Comment (sm3142) [inline docs/docs/how-to/oauth2-jwt.md:90]:
```suggestion
> So far, the code above is only documenting the authorization scheme and required scopes, but does not actually authorize incoming requests. The next section will explain how to achieve the latter.
```
typo

**Comment by danielgtaylor** [comment]
@sm3142 @Insei sorry for the delay. I caught covid and have been pretty sick. Thanks for all the help, I've applied all the suggestions and will merge this in, but feel free to open additional PRs if you think anything else should get added or updated 👍🏻

**Comment by sm3142** [comment]
@danielgtaylor, hope you feel better now. Thanks for all the effort!

**Pull request merged**